js/bitstamp.js

```diff
@@ -993,26 +993,44 @@ module.exports = class bitstamp extends Exchange {
     }
 
     handleErrors (httpCode, reason, url, method, headers, body, response) {
-        if (typeof body !== 'string')
-            return; // fallback to default error handler
-        if (body.length < 2)
-            return; // fallback to default error handler
-        if ((body[0] === '{') || (body[0] === '[')) {
-            // fetchDepositAddress returns {"error": "No permission found"} on apiKeys that don't have the permission required
-            let error = this.safeString (response, 'error');
-            let exceptions = this.exceptions;
-            if (error in exceptions) {
-                throw new exceptions[error] (this.id + ' ' + body);
+        if (response === undefined) {
+            return;
+        }
+        // fetchDepositAddress returns {"error": "No permission found"} on apiKeys that don't have the permission required
+        let status = this.safeString (response, 'status');
+        let error = this.safeValue (response, 'error');
+        if (status === 'error' || error) {
+            let errors = [];
+            if (typeof error !== 'string') {
+                let keys = Object.keys (error);
+                for (let i = 0; i < keys.length; i++) {
+                    let thisError = error[keys[i]];
+                    if (Array.isArray (thisError)) {
+                        thisError = thisError.join ('. ');
+                    }
+                    errors.push (keys[i] + ': ' + thisError);
+                }
+            } else {
+                errors.push (error);
             }
-            let status = this.safeString (response, 'status');
-            if (status === 'error') {
-                let code = this.safeString (response, 'code');
-                if (code !== undefined) {
-                    if (code === 'API0005')
-                        throw new AuthenticationError (this.id + ' invalid signature, use the uid for the main account if you have subaccounts');
+            if (response.reason && response.reason['__all__'] && Array.isArray (response.reason['__all__'])) {
+                for (let i = 0; i < response.reason['__all__'].length; i++) {
+                    errors.push (response.reason['__all__'][i]);
                 }
-                throw new ExchangeError (this.id + ' ' + body);
             }
+            let exceptions = this.exceptions;
+            for (let i = 0; i < errors.length; i++) {
+                if (errors[i] in exceptions) {
+                    throw new exceptions[error] (this.id + ': ' + errors[i]);
```

Comment:
This looks suspicious – `if (errors[i] in exceptions)` we shouldn't throw `exceptions[error]`, but should probably throw

```JavaScript
const e = errors[i];
if (e in exceptions) {
    throw new exceptions[e] (...
}
```

right?

Reply:
Ah sorry, yes I made that mistake on a few lines down and managed to catch it. Missed this one!

Reply:
@devsi no worries, we'll polish it ) Can you also post a sample of their response with multiple errors in it here? I'll use it for testing on my side.

Reply:
Sure: A good test for me was to try and withdraw with a false address and not enough funds. So:

`exchange.withdraw('BTC', 0.01, 'lskdmgnlkmt');`

and I get : 

```
[
  "error": [
    "amount": [
      0 => "You have only 0.00100000 BTC available. Check your account balance for details."
    ]
    "address":  [
      0 => "Ensure this value has at least 25 characters (it has 4)."
    ]
  ]
]
```

I can't replicate an error with multiple results in just "amount" though for example. I was able to intercept the response and add an additional fake error in there to test.